src/components/student/student.controller.ts

```diff
@@ -0,0 +1,94 @@
+import Student from "../student/student.model";
+import jwt = require("jsonwebtoken");
+/**
+ * Controller for managing student-related operations.
+ */
+class StudentController {
+  /**
+   * Take attendance for the logged-in student.
+   *
+   * @param req - The request object.
+   * @param res - The response object.
+   * @param next - The next function in the middleware chain.
+   */
+  async takeAttendence(req, res, next) {
+    try {
+      const student = req.user;
+      console.log(student);
+      const currentDate = new Date().toISOString().split("T")[0]; // Format: "YYYY-MM-DD"
```

Comment:
suggestion: here you can apply dayjs library **[Link](https://www.npmjs.com/package/dayjs)** for reference please check